### PR TITLE
tests: add podman desktop playwright tests dependency

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
Introduces the dependency on `@podman-desktop/tests-playwright` e2e testing library for podman desktop.
### What issues does this PR fix or reference?
#198 

### How to test this PR?
```
git clone podman-desktop
cd podman-desktop
yarn install; yarn test:e2e:build
export PODMAN_DESKTOP_ARGS=$(pwd)
cd ..
cd podman-desktop-extension-bootc
yarn install
yarn test:e2e
```
Now the e2e tests from bootc extension should be executed and use podman-desktop that is locally built from codebase.
check `tests/output` folder for the tests results.

### CODE REVIEW

**Comment by odockal** [comment]
The PR requires npm package available. This PR needs to be finished and merged: https://github.com/containers/podman-desktop/pull/6442

**Review by benoitf — COMMENTED** [review]
I'm wondering if we could avoid to have the vitest.config.js and extended-hook in this repository ?

use a vitest.config.js provided by the playwright package ?

File: package.json [hunk 1]
```diff
@@ -22,17 +22,24 @@
...
+    "@podman-desktop/tests-playwright": "^0.0.202403201042-07a9341",
     "@typescript-eslint/eslint-plugin": "^7.0.0",
     "@typescript-eslint/parser": "^6.21.0",
     "@vitest/coverage-v8": "^1.2.2",
     "autoprefixer": "^10.4.17",
     "concurrently": "^8.2.2",
+    "electron": "29.1.4",
+    "electron-builder": "23.6.0",
+    "electron-builder-notarize": "^1.5.1",
+    "electron-devtools-installer": "^3.2.0",
```
[the remark this answers is not in the document]
Comment (benoitf) [reply package.json:42]: I think tests folder should have its own package.json file

**Comment by benoitf** [comment]
include a package.json in tests folder (also maybe use a subfolder like tests/e2e or rename tests to e2e I don't know)

and include the tests in the yarn workspace https://github.com/containers/podman-desktop-extension-bootc/blob/b4c0bb982eb72f27c8af56802c8da6d47b54439b/package.json#L61

all the dependencies used for e2e should not be leaking in the root package.json

**Comment by odockal** [comment]
> I'm wondering if we could avoid to have the vitest.config.js and extended-hook in this repository ?
> 
> use a vitest.config.js provided by the playwright package ?

good enhancements: https://github.com/containers/podman-desktop/issues/6478 and https://github.com/containers/podman-desktop/issues/6477

**Comment by odockal** [comment]
@benoitf @cbr7 I have update the PR with required changes. I have run it locally also:
```
 ✓ src/bootc-extension.spec.ts (13) 80592ms
   ✓ BootC Extension (13) 71895ms
     ✓ Go to settings and check if extension is already installed 1251ms
     ↓ Uninstalled previous version of bootc extension [skipped]
     ✓ Install extension through Settings 6746ms
     ✓ Build bootc image from containerfile 61798ms
     ↓ Building bootable image type: QCOW2 for architecture: ARM64 [skipped]
     ↓ Building bootable image type: QCOW2 for architecture: AMD64 [skipped]
     ↓ Building bootable image type: AMI for architecture: ARM64 [skipped]
     ↓ Building bootable image type: AMI for architecture: AMD64 [skipped]
     ↓ Building bootable image type: RAW for architecture: ARM64 [skipped]
     ↓ Building bootable image type: RAW for architecture: AMD64 [skipped]
     ↓ Building bootable image type: ISO for architecture: ARM64 [skipped]
     ↓ Building bootable image type: ISO for architecture: AMD64 [skipped]
     ✓ Remove bootc extension through Settings 2097ms

 Test Files  1 passed (1)
      Tests  4 passed | 9 skipped (13)
   Start at  16:17:16
   Duration  81.30s (transform 109ms, setup 346ms, collect 14ms, tests 80.59s, environment 0ms, prepare 93ms)

Done in 82.26s.
```

If someone can please try the steps to verify that all of this work. Next PR could be for adding CI for some of the tests.

**Comment by odockal** [comment]
I got some tests run from my fork with proposed workflow: https://github.com/odockal/podman-desktop-extension-bootc/actions/runs/8363989131/job/22898241179

File: tests/playwright/package.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  "description": "Podman Desktop BootC extension Playwright E2E tests",
+  "scripts": {
+    "test:e2e:setup": "xvfb-maybe --auto-servernum --server-args='-screen 0 1280x960x24' --",
+    "test:e2e": "npm run test:e2e:setup vitest run src/ --pool=threads --poolOptions.threads.singleThread --poolOptions.threads.isolate --no-file-parallelism"
+  },
+  "author": "Red Hat",
+  "license": "Apache-2.0",
+  "dependencies": {
+    "@podman-desktop/tests-playwright": "^0.0.202403202143-15b0f59"
+  },
```
Comment (benoitf) [inline tests/playwright/package.json:13]: remove as it's already in devDependencies
```suggestion
```

File: tests/playwright/resources/bootable-containerfile [hunk 3]
```diff
@@ -0,0 +1,4 @@
+FROM quay.io/centos-bootc/fedora-bootc:eln
```
Comment (benoitf) [inline tests/playwright/resources/bootable-containerfile:1]: I think I commented somewhere else, it's missing the Copyright/header and the newline at the bottom

File: tests/playwright/src/setupFiles/extended-hooks.ts [hunk 4]
```diff
@@ -0,0 +1,24 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
```
Comment (benoitf) [inline tests/playwright/src/setupFiles/extended-hooks.ts:2]:
```suggestion
 * Copyright (C) 2024 Red Hat, Inc.
```

File: tests/playwright/vitest.config.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
```
Comment (benoitf) [inline tests/playwright/vitest.config.js:2]:
```suggestion
 * Copyright (C) 2024 Red Hat, Inc.
```

File: tests/playwright/package.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+  "devDependencies": {
+    "@playwright/test": "^1.42.1",
+    "@podman-desktop/tests-playwright": "^0.0.202403202143-15b0f59",
+    "@types/node": "^20",
+    "electron": "^29.1.4",
```
Comment (benoitf) [inline tests/playwright/package.json:18]: not a blocker for this PR, but here probably we should add that as a dependency on `@podman-desktop/tests-playwright"` side so we don't need to enforce that there

Comment (odockal) [reply tests/playwright/package.json:15]: Just `electron` or all?

File: tests/playwright/resources/bootable-containerfile (hunk 3 above)
Comment (odockal) [reply tests/playwright/resources/bootable-containerfile:1]: We are missing headers in podman-desktop, might be there.

File: tests/playwright/package.json (hunk 6 above)
Comment (benoitf) [reply tests/playwright/package.json:18]: I would say for all dependencies that we're not using from this project

for example when writing a spec.ts file you're using

```
import type { Locator, Page } from '@playwright/test';
import { afterAll, beforeAll, test, describe, beforeEach, expect } from 'vitest';
```

so anyway you need to depend on playwright and vitest so these one are kind of mandatory
but for electron as we don't have direct import on it, it should be transitive

**Comment by odockal** [comment]
@benoitf I have committed your suggestion to the PR directly, but these got overridden by my commit not having your suggestions. How do I pull commits from PR into my branch?

**Review by benoitf — APPROVED** [review]

File: tests/playwright/package.json (hunk 6 above)
Comment (odockal) [reply tests/playwright/package.json:15]: I see.  That makes perfect sense.thx!

**Comment by benoitf** [comment]
in your local branch you need first to pull remote changes

assuming your fork is named 'odockal' and `origin` the current repository

pull first applied changes using the UI
```
git pull --rebase odockal tests-playwright-deps
```

then you have the changes locally on your machine

but maybe I've not understood what you wanted

**Review by deboer-tim — APPROVED** [review]

**Comment by benoitf** [comment]
@odockal if the remark is that I'm not attributed some contribution I don't care as long as the fix is there

**Comment by odockal** [comment]
> in your local branch you need first to pull remote changes
> 
> assuming your fork is named 'odockal' and `origin` the current repository
> 
> pull first applied changes using the UI
> 
> ```
> git pull --rebase odockal tests-playwright-deps
> ```
> 
> then you have the changes locally on your machine
> 
> but maybe I've not understood what you wanted

Yes I think this is what I wanted. Thx.



> @odockal if the remark is that I'm not attributed some contribution I don't care as long as the fix is there

Sure, no problem.

**Pull request merged**